src/main/java/org/apache/maven/plugin/compiler/AbstractCompilerMojo.java

```diff
@@ -330,6 +335,22 @@ public abstract class AbstractCompilerMojo extends AbstractMojo {
     @Parameter
     private List<DependencyCoordinate> annotationProcessorPaths;
 
+    /**
+     * <p>
+     * Whether to use the Maven dependency management section when resolving transitive dependencies of annotation
+     * processor paths.
+     * </p>
+     * <p>
+     * This flag does not enable / disable the ability to get the version of annotation processor paths
+     * (the top-level paths) from dependency management section. It only influences the resolution of
+     * transitive dependencies of those top-level paths.
+     * </p>
+     *
+     * @since 3.12.0
+     */
+    @Parameter(defaultValue = "false")
+    private boolean annotationProcessorPathsUseDepMgmt;
```

Comment:
I am not the biggest fan of this long name, but I could not find a better one at this point. Reason I choose this one is that the prefix `annotationProcessorPaths` means it will be rendered/shown together with the paths option, meaning users are much more likely to see this when browsing the docs (and making the connection that is used for annotation processor paths). We could use something like `useDepMgmt` or so, but then you need to know exactly what you are looking for, since this would be rendered very far away from the `annotationProcessorPaths`. Also something like `useDepMgmt` may be too generic?